clang/test/SemaCXX/gh102293.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clang_cc1 -std=c++23 -fsyntax-only -verify %s
+// expected-no-diagnostics
+
+template <typename T> static void destroy() {
+    T t;
+    ++t;
+}
+
+struct Incomplete;
+
+template <typename = int> struct HasD {
+  ~HasD() { destroy<Incomplete*>(); }
+};
+
+struct HasVT {
+  virtual ~HasVT();
+};
+
+struct S : HasVT {
+  HasD<> v;
+};
+
```

Comment:
What happen if you do 

`static_assert((S{}, 1));`

(and do we care that it's probably going to complain about the destructor nit being defined?)

Reply:
I don't think there NO tests checking that complain?